File: lua/seed_audio_ai_site_kit.lua
local M = {}

M.site = {
  name = "Seed Audio AI",
  homepage = "https://seedaud.io/",
  description = "Browser-based text-to-speech and AI voice generation workflow.",
  canonicalPages = {
    home = "https://seedaud.io/",
    textToSpeech = "https://seedaud.io/text-to-speech/",
    pricing = "https://seedaud.io/pricing/",
    safety = "https://seedaud.io/safety/",
    terms = "https://seedaud.io/terms/"
  },
  tags = { "text-to-speech", "ai-voice", "voice-generator", "site-kit" }
}

function M.seed_audio_url(path)
  local clean = tostring(path or ""):gsub("^/+", ""):gsub("/+$", "")
  if clean == "" then
    return M.site.homepage
  end
  return M.site.homepage .. clean .. "/"
end

return M

File: elixir/lib/seed_audio_ai_site_kit.ex
defmodule SeedAudioAISiteKit do
  @moduledoc """
  Unofficial metadata and URL helpers for Seed Audio AI at https://seedaud.io/.
  """

  @homepage "https://seedaud.io/"

  def site do
    %{
      name: "Seed Audio AI",
      homepage: @homepage,
      description: "Browser-based text-to-speech and AI voice generation workflow.",
      canonical_pages: %{
        home: @homepage,
        text_to_speech: "https://seedaud.io/text-to-speech/",
        pricing: "https://seedaud.io/pricing/",
        safety: "https://seedaud.io/safety/",
        terms: "https://seedaud.io/terms/"
      },
      tags: ["text-to-speech", "ai-voice", "voice-generator", "site-kit"]
    }
  end

  def seed_audio_url(path \\ "") do
    clean = path |> to_string() |> String.trim("/")
    if clean == "", do: @homepage, else: @homepage <> clean <> "/"
  end
end
